crates/node/src/init.rs

```diff
@@ -30,7 +31,7 @@ enum JoinError {
     MissingNodesConfiguration,
     #[error("detected a concurrent registration for node '{0}'")]
     ConcurrentNodeRegistration(String),
-    #[error("failed writing to metadata store: {0}")]
+    #[error("failed writing to metadata store: {:#}", .0.decorate())]
```

Comment:
I personally don't like using the pretty print here but it's the only way to get this to show up when the error is printed since the type is erased (via anyhow).

Reply:
I am wondering whether it would be ok if we make the error description a bit more coarse grained and make the `JoinError` implement the `CodedError`. Then we could also add some error codes for the other join errors that terminate the join attempt. For the `MetadataStore` variant, the description would be admittedly a bit more generic (like failed writing to the metadata store, make sure that you've configured your cluster correctly...).

Reply:
One problem we will have with `CodedError` at the moment is that the `TaskCenter` which logs the error that bubbles up from the task, isn't aware of `CodedErrors`. Often, we also wrap those errors into `anyhow::Error` which hides it even further. Maybe in this case, it is ok to not print the description. Users will be able to look it up online.